src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -2078,4 +2084,142 @@ public Mono<Void> updateLineSplitWithVoltageLevel(UUID modificationUuid, LineSpl
+                .thenMany(getNetworkModificationInfos(networkUuid, variantId)
+                .flatMapIterable(networkInfos -> {
+                    NetworkStoreListener listener = NetworkStoreListener.create(networkInfos.getNetwork(), networkUuid, groupUuid, networkModificationRepository, equipmentInfosService, false, networkInfos.isApplyModifications());
+                    ReporterModel reporter = new ReporterModel(NETWORK_MODIFICATION_REPORT_KEY, NETWORK_MODIFICATION_REPORT_NAME);
+                    Reporter subReporter = reporter.createSubReporter("lineAttachToVoltageLevel", "Attach line to voltage level");
+
+                    return execLineAttachToVoltageLevel(listener, lineAttachToVoltageLevelInfos, reportUuid);
+                }));
+    }
+
+    public Mono<Void> updateLineAttachToVoltageLevel(UUID modificationUuid, LineAttachToVoltageLevelInfos lineAttachToVoltageLevelInfos) {
+        assertLineAttachToVoltageLevelInfosNotEmpty(lineAttachToVoltageLevelInfos).subscribe();
+
+        Optional<ModificationEntity> lineAttachToVoltageLevelEntity = this.modificationRepository.findById(modificationUuid);
+
+        if (lineAttachToVoltageLevelEntity.isEmpty()) {
+            return Mono.error(new NetworkModificationException(LINE_ATTACH_NOT_FOUND, "Line attach not found"));
+        }
+
+        LineAttachToVoltageLevelEntity casted = (LineAttachToVoltageLevelEntity) lineAttachToVoltageLevelEntity.get();
+        VoltageLevelCreationEntity mayVoltageLevelCreation = casted.getMayVoltageLevelCreation();
+        VoltageLevelCreationInfos mayNewVoltageLevelInfos = lineAttachToVoltageLevelInfos.getMayNewVoltageLevelInfos();
+        LineCreationEntity lineCreation = casted.getLineCreation();
+        LineCreationInfos lineInfos = lineAttachToVoltageLevelInfos.getAttachmentLine();
+
+        LineAttachToVoltageLevelEntity updatedEntity = LineAttachToVoltageLevelEntity.toEntity(
+                lineAttachToVoltageLevelInfos.getLineToAttachToId(),
+                lineAttachToVoltageLevelInfos.getPercent(),
+                lineAttachToVoltageLevelInfos.getAttachmentPointId(),
+                lineAttachToVoltageLevelInfos.getAttachmentPointName(),
+                mayNewVoltageLevelInfos,
+                lineAttachToVoltageLevelInfos.getExistingVoltageLevelId(),
+                lineAttachToVoltageLevelInfos.getBbsOrBusId(),
+                lineInfos,
+                lineAttachToVoltageLevelInfos.getNewLine1Id(),
+                lineAttachToVoltageLevelInfos.getNewLine1Name(),
+                lineAttachToVoltageLevelInfos.getNewLine2Id(),
+                lineAttachToVoltageLevelInfos.getNewLine2Name()
+        );
+        updatedEntity.setId(modificationUuid);
+        updatedEntity.setGroup(lineAttachToVoltageLevelEntity.get().getGroup());
+        this.networkModificationRepository.updateModification(updatedEntity);
+
+        // NetworkStoreListener.makeVoltageLevelCreationEntity recreates on need, so get rid of previous
```

Comment:
I don't understand why we need to delete objects here, does it mean we are inserting unwanted data when updating LineAttachment ?

Reply:
In case the attributes of the created voltage level or the created line are modified when updating line attachment creation, the older attributes have to be deleted since new ones will be automatically created.